### PR TITLE
Fix hardhat service

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

Fixes the hardhat service configuration assuming that we are using the registries image in the deployment.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [x] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

n/a

### CODE REVIEW

File: packages/valory/services/oracle_hardhat/service.yaml [hunk 1]
```diff
@@ -14,10 +14,16 @@ extra:
...
+        - - '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266'
+          - '0x70997970C51812dc3A010C7d01b50e0d17dc79C8'
+          - '0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC'
+          - '0x90F79bf6EB2c4f870365E785982E1f101E93b906'
+        safe_contract_address:
+        - '0xD8dE647170163a981bb3Fdb2063583eAcF7D55AC'
       observation_interval: 30
       broadcast_to_server: false
-      service_registry_address: '0x0DCd1Bf9A1b36cE34237eEaFef220932846BCD82'
+      service_registry_address: '0x998abeb3E57409262aE5b751f60747921B33613E'
```
Comment (Adamantios) [inline packages/valory/services/oracle_hardhat/service.yaml:26]: The address in the registries image.

File: packages/valory/services/oracle_hardhat/service.yaml [hunk 2]
```diff
@@ -14,10 +14,16 @@ extra:
   setup_args:
     args: &id001
       setup:
-        safe_contract_address: []
+        all_participants:
+        - - '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266'
+          - '0x70997970C51812dc3A010C7d01b50e0d17dc79C8'
+          - '0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC'
+          - '0x90F79bf6EB2c4f870365E785982E1f101E93b906'
```
Comment (Adamantios) [inline packages/valory/services/oracle_hardhat/service.yaml:21]: When deploying, the `keys.json` needs to contain these addresses and their corresponding private keys. These are the ones in the registries image for service id 1.

Maybe we would need to update the keys.json suggestion in the documentation to make this easier for the users @jmoreira-valory.

File: packages/valory/services/oracle_hardhat/service.yaml [hunk 3]
```diff
@@ -14,10 +14,16 @@ extra:
...
+        safe_contract_address:
+        - '0xD8dE647170163a981bb3Fdb2063583eAcF7D55AC'
```
Comment (Adamantios) [inline packages/valory/services/oracle_hardhat/service.yaml:23]: The address in the registries image.

File: packages/valory/services/oracle_hardhat/service.yaml [hunk 4]
```diff
@@ -117,7 +123,7 @@ type: connection
 config:
   ledger_apis:
     ethereum:
-      address: ${SERVICE_ORACLE_HARDHAT_RPC:str:http://host.docker.internal:8545}
+      address: ${SERVICE_ORACLE_HARDHAT_RPC:str:http://hardhat:8545}
```
Comment (Adamantios) [inline packages/valory/services/oracle_hardhat/service.yaml:126]: Should be the name of the registries image in the docker-compose file.

**Review by 0xArdi — APPROVED** [review]

File: packages/valory/services/oracle_hardhat/service.yaml (hunk 2 above)
Comment (jmoreira-valory) [reply packages/valory/services/oracle_hardhat/service.yaml:21]: Please, update the keys and addresses in the index.md file in this PR. I will review the text in another PR.

Comment (Adamantios) [reply packages/valory/services/oracle_hardhat/service.yaml:21]: To save some CI time, could you edit them on the PR you mentioned? They should be:

```json
[
  {
      "address": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
      "private_key": "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
  },
  {
      "address": "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
      "private_key": "0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d"
  },
  {
      "address": "0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC",
      "private_key": "0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a"
  },
  {
      "address": "0x90F79bf6EB2c4f870365E785982E1f101E93b906",
      "private_key": "0x7c852118294e51e653712a81e05800f419141751be58f605c371e15141b007a6"
  }
]
```

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**